### PR TITLE
ENH: add support for na_filter in Python engine

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

### CODE REVIEW

**Comment by jreback** [comment]
looks fine. pls rebase

**Comment by gfyoung** [comment]
Rebased and Travis is giving the green light.  Ready to merge if there are no other concerns.

**Comment by gfyoung** [comment]
@jreback : Rebased and Travis is giving the green light.  Ready to merge if there are no other concerns.

File: doc/source/whatsnew/v0.18.2.txt [hunk 1]
```diff
@@ -75,6 +75,7 @@ Other enhancements
     pd.Timestamp(year=2012, month=1, day=1, hour=8, minute=30)
 
 - The ``pd.read_csv()`` with ``engine='python'`` has gained support for the ``decimal`` option (:issue:`12933`)
+- The ``pd.read_csv()`` with ``engine='python'`` has gained support for the ``na_filter`` option (:issue:`13321`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.18.2.txt]: Can you combine this with the previous point in one sentence?

Comment (gfyoung) [inline doc/source/whatsnew/v0.18.2.txt]: I thought we leave separate issues on separate lines?

**Comment by jorisvandenbossche** [comment]
@gfyoung very minor comment

File: doc/source/whatsnew/v0.18.2.txt (hunk 1 above)
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.18.2.txt]: Not that important! (and I don't know if we have a rule about it)

Comment (gfyoung) [inline doc/source/whatsnew/v0.18.2.txt]: Fair enough.  Unless there are more objections about this, I'll leave as is.

Comment (jreback) [inline doc/source/whatsnew/v0.18.2.txt]: I think we try to have separate issue on separate lines that were closed by different PR's, but yes occasionally do bunch them up if they are repetitive.

Comment (gfyoung) [inline doc/source/whatsnew/v0.18.2.txt]: So...how about this one?

**Comment by jreback** [comment]
ty @gfyoung
